k8s/src/main/scala/io/buoyant/k8s/EndpointsNamer.scala

```diff
@@ -49,12 +50,67 @@ class EndpointsNamer(
         Activity.value(NameTree.Neg)
     }
   }
+}
+
+class SingleNsNamer(
+  idPrefix: Path,
+  labelName: Option[String],
+  nsName: String,
+  mkApi: String => NsApi,
+  backoff: Stream[Duration] = Backoff.exponentialJittered(10.milliseconds, 10.seconds)
+)(implicit timer: Timer = DefaultTimer.twitter) extends EndpointsNamer(idPrefix, mkApi, backoff)(timer) {
+
+  val PrefixLen = 2
+  private[this] val variablePrefixLength = PrefixLen + labelName.size
+
+  /**
+   * Accepts names in the form:
+   *   /<port-name>/<svc-name>/residual/path
```

Comment:
Maybe add a note here and in the scaladoc for the multins lookup method about the optional label segment.